tests/TrustedHostsNetworkResolverTest.php

```diff
@@ -22,148 +22,153 @@ public function dataProcessTrusted(): array
-                [
-                    ['hosts' => ['5.5.5.5', '2.2.2.2', '127.0.0.1'], 'ipHeaders' => ['x-forwarded-for']],
-                ],
-                '2.2.2.2',
-            ],
+            'hosts: server IP' => [[
+                'remoteAddr' => '18.18.18.18',
+                'trustedHosts' => [
+                    ['hosts' => ['18.18.18.18'], 'ipHeaders' => ['x-forwarded-for']],
+                ],
+                'headers' => ['x-forwarded-for' => ['9.9.9.9', '5.5.5.5', '2.2.2.2']],
+                'expectedClientIp' => null,
+            ]],
+            'hosts: server IP, with first one proxy, ipHeaders: not set' => [[
+                'remoteAddr' => '18.18.18.18',
+                'trustedHosts' => [
+                    ['hosts' => ['2.2.2.2', '18.18.18.18'], 'ipHeaders' => []],
+                ],
+                'headers' => ['x-forwarded-for' => ['9.9.9.9', '5.5.5.5', '2.2.2.2']],
+                'expectedClientIp' => '18.18.18.18',
+            ]],
+            'hosts: server IP with first one proxy' => [[
+                'remoteAddr' => '18.18.18.18',
+                'trustedHosts' => [
+                    ['hosts' => ['2.2.2.2', '18.18.18.18'], 'ipHeaders' => ['x-forwarded-for']],
+                ],
+                'headers' => ['x-forwarded-for' => ['9.9.9.9', '5.5.5.5', '2.2.2.2']],
+                'expectedClientIp' => '5.5.5.5',
+            ]],
+            'hosts: server IP with first few proxies' => [[
+                'remoteAddr' => '18.18.18.18',
+                'trustedHosts' => [
+                    ['hosts' => ['5.5.5.5', '2.2.2.2', '18.18.18.18'], 'ipHeaders' => ['x-forwarded-for']],
+                ],
+                'headers' => ['x-forwarded-for' => ['9.9.9.9', '8.8.8.8', '5.5.5.5', '2.2.2.2']],
+                'expectedClientIp' => '8.8.8.8',
+            ]],
 
             // Port headers
 
-            'port headers, separate header value' => [
-                ['x-forwarded-for' => ['1234']],
-                [
-                    ['hosts' => ['172.16.0.1', '127.0.0.1'], 'ipHeaders' => [], 'portHeaders' => ['x-forwarded-for']],
-                ],
-                '127.0.0.1',
-                '',
-                'http',
-                '/',
-                '',
-                1234,
-            ],
-            'port headers, provided with proxy' => [
-                [
-                    'x-forwarded-for' => ['1234'],
-                    'forwarded' => ['for=9.9.9.9', 'proto=http;for="5.5.5.5:4321";host=test', 'for=2.2.2.2'],
-                ],
-                [
+            'port headers, separate header value' => [[
+                'remoteAddr' => '18.18.18.18',
+                'trustedHosts' => [
+                    ['hosts' => ['172.16.0.1', '18.18.18.18'], 'ipHeaders' => [], 'portHeaders' => ['x-forwarded-for']],
+                ],
+                'headers' => ['x-forwarded-for' => ['1234']],
```

Comment:
Using `X-Forwarded-For` for just a port is very confusing.

```suggestion
                'trustedHosts' => [
                    ['hosts' => ['172.16.0.1', '18.18.18.18'], 'ipHeaders' => [], 'portHeaders' => ['x-forwarded-port']],
                ],
                'headers' => ['x-forwarded-port' => ['1234']],
```